pkg/common/scheduler/statefulset/state.go

```diff
@@ -112,7 +130,31 @@ func (s *stateBuilder) State() (*state, error) {
 			}
 		}
 	}
-	return &state{free: free, lastOrdinal: last, capacity: s.capacity}, nil
+
+	if s.schedulerPolicy == EvenSpread {
+		//TODO: need a node watch to see if # nodes/ # zones have gone up or down
+		nodes, err := kubeclient.Get(s.ctx).CoreV1().Nodes().List(s.ctx, metav1.ListOptions{})
```

Comment:
We should use a NodeLister here to hit the cache.

Reply:
Scheduler and state now using `node lister` and `pod lister` now to get zone info from all nodes, and also to get nodename from pod to compute its zone. 

Only piece remaining as part of this refactoring are the three test files (state_test, autoscaler_test and scheduler_test) that do create dummy nodes, pods etc and need to pass a node lister instance from the test context to the state accessor. Unsure of how to use the new method [nodelister()](https://github.com/knative-sandbox/eventing-kafka/pull/587/files#diff-376bd4cf279c82c42a8d2d1e7b070dd370c89f8148948b230822e8ddd94b8dcfR216) in all three of my testcase files  without a [Listers](https://github.com/knative/eventing/blob/main/pkg/reconciler/testing/v1/listers.go#L71) object that may need some setup with reconciler testing methods instead of what we have now.... @lionelvillard will need to check with you on this.

Reply:
Done. Changes in latest test files